CH32V003  / all WCH socs where RCC_PLLMULL is not defined 's build is broken

### Describe the bug

Recent merges changed how the board RCC is managed.
this causes RCC_PLLMULL to be present in the code despite not being defined for all SOCs, causing a build failure for ch32v003evt and others.
Somehow this was not caught by CI

### Regression

- [ ] This is a regression.

### Steps to reproduce

`west build -p -c ch32v003evt samples/hellow_world`

### Relevant log output

```shell
/home/vdragon/dev/zephyr_upstreaming/zephyr/drivers/clock_control/clock_control_wch_rcc.c: In function 'clock_control_wch_rcc_init':
/home/vdragon/dev/zephyr_upstreaming/zephyr/drivers/clock_control/clock_control_wch_rcc.c:176:32: error: 'RCC_PLLMULL' undeclared (first use in this function)
  176 |                 RCC->CFGR0 &= ~RCC_PLLMULL;
      |                                ^~~~~~~~~~~
/home/vdragon/dev/zephyr_upstreaming/zephyr/drivers/clock_control/clock_control_wch_rcc.c:176:32: note: each undeclared identifier is reported only once for each function it appears in
```

### Impact

Showstopper – Prevents release or major functionality; system unusable.

### Environment

latest mainline

### Additional Context

fix by adding
```c
#ifndef RCC_PLLMULL
#define RCC_PLLMULL 0xBEEF
#endif
```
at the top or something better.

--- a/drivers/clock_control/clock_control_wch_rcc.c
+++ b/drivers/clock_control/clock_control_wch_rcc.c
@@ -35,6 +35,7 @@
 #define WCH_RCC_SRC_IS_HSI 1
 #endif
 
+#if defined(CONFIG_DT_HAS_WCH_CH32V20X_30X_PLL_CLOCK_ENABLED)
 #if defined(CONFIG_SOC_CH32V307)
 /* TODO: Entry 13 is 6.5x (fractional multiple currently unsupported without
  * changes to RCC config datatype)
@@ -42,6 +43,7 @@
 static const uint8_t pllmul_lut[] = {18, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 0, 15, 16};
 #else
 static const uint8_t pllmul_lut[] = {2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 18};
+#endif
 #endif
 
 struct clock_control_wch_rcc_config {
@@ -165,7 +167,7 @@ static int clock_control_wch_rcc_init(const struct device *dev)
 		} else if (IS_ENABLED(WCH_RCC_PLL_SRC_IS_HSI)) {
 			RCC->CFGR0 &= ~RCC_PLLSRC;
 		}
-
+#if defined(RCC_PLLMULL)
 		uint8_t pllmul = 0x0; /* Default Reset Value */
 
 		for (size_t i = 0; i < ARRAY_SIZE(pllmul_lut); i++) {
@@ -175,6 +177,7 @@ static int clock_control_wch_rcc_init(const struct device *dev)
 		}
 		RCC->CFGR0 &= ~RCC_PLLMULL;
 		RCC->CFGR0 |= WCH_RCC_PLLMUL_VAL(pllmul);
+#endif
 		RCC->CTLR |= RCC_PLLON;
 		while ((RCC->CTLR & RCC_PLLRDY) == 0) {
 		}

